pkg/utils/k8s.go

```diff
@@ -205,14 +219,128 @@ func NamespaceDoesNotExist(config *rest.Config, namespace string) error {
 		return err
 	}
+
 	return nil
 }
+
+// SecretDoesNotExist returns no error only if the specified secret does not exist in the k8s cluster
+func SecretDoesNotExist(config *rest.Config, name, namespace string) error {
+	_, err := GetSecret(config, name, namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	return fmt.Errorf("secret %v exists in cluster", namespace)
+}
+
+// SecretExists returns no error only if the specified secret exists in the k8s cluster
+func SecretExists(config *rest.Config, name, namespace string) error {
+	_, err := GetSecret(config, name, namespace)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetStorageOSCluster returns the storageoscluster object if it exists in the k8s cluster.
+// Use 'List' to discover as there can only be one object per k8s cluster and 'List' does not
+// require name/namespace - input namespace can optionally be passed to narrow the search.
+func GetStorageOSCluster(config *rest.Config, namespace string) (operatorapi.StorageOSCluster, error) {
+	scheme := runtime.NewScheme()
+	operatorapi.AddToScheme(scheme)
+	stosCluster := operatorapi.StorageOSCluster{}
+
+	newClient, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		return stosCluster, err
+	}
+	stosClusterList := &operatorapi.StorageOSClusterList{}
+	listOption := (&client.ListOptions{}).ApplyOptions([]client.ListOption{client.InNamespace(namespace)})
+	err = newClient.List(context.TODO(), stosClusterList, listOption)
+	if err != nil {
+		return stosCluster, err
+	}
+
+	if len(stosClusterList.Items) == 0 {
+		return stosCluster, errors.NewNotFound(operatorapi.SchemeGroupVersion.WithResource("StorageOSCluster").GroupResource(), "")
+	}
+
+	stosCluster = stosClusterList.Items[0]
+	return stosCluster, nil
+}
+
+// StorageOSClusterDoesNotExist return no error only if no storageoscluster object exists in k8s cluster
+func StorageOSClusterDoesNotExist(config *rest.Config, namespace string) error {
+	_, err := GetStorageOSCluster(config, namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	return fmt.Errorf("storageoscluster exists")
+}
+
+// GetEtcdCluster returns the etcdcluster object of name and namespace.
+func GetEtcdCluster(config *rest.Config, name, namespace string) (*etcdoperatorapi.EtcdCluster, error) {
+	scheme := runtime.NewScheme()
+	etcdoperatorapi.AddToScheme(scheme)
+	etcdCluster := &etcdoperatorapi.EtcdCluster{}
+
+	newClient, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		return etcdCluster, err
+	}
+	err = newClient.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, etcdCluster)
+	if err != nil {
+		return etcdCluster, err
+	}
+	return etcdCluster, nil
+}
+
+// EtcdClusterDoesNotExist return no error only if no etcdcluster object exists in k8s cluster
+func EtcdClusterDoesNotExist(config *rest.Config, name, namespace string) error {
```

Comment:
Please do not modify the function (or the others). This is just a chat, what do you think.

```
func EtcdClusterDoesNotExist(config *rest.Config, name, namespace string) error {
	_, err := GetEtcdCluster(config, name, namespace)
	if err == nil {
            return fmt.Errorf("etcdcluster exists")
        }
	if !errors.IsNotFound(err) {
	    return err
        }

	return nil
}
```

In this way happy path goes to the left as possible. There is no chained if statements. And the shape is more nicer it looks a great triangle :D.
I prefer this style because it is more goish and easier to parse with eyes.